app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/utils.ts

```
@@ -35,7 +35,7 @@ export function getNumericDifference(
 export function formatValue(value: string | number, fractionDigits = 3) {
   const numericValue =
     typeof value === "number" ? value : parseFloat(value as string);
-  if (!isNaN(numericValue)) {
+  if (!isNaN(numericValue) && numericValue == value) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)